simple_settings/dynamic_settings/base.py

```diff
@@ -30,9 +31,19 @@ def set(self, key, value):
             return
         if self.auto_casting:
             value = jsonpickle.encode(value)
-        self._set(key, value)
+        self._set(self._qualified_key(key), value)
 
     def _is_valid_key(self, key):
         if not self.key_pattern:
             return True
         return bool(re.match(self.key_pattern, key))
+
+    def _qualified_key(self, key):
+        """
+        Prepends the configured prefix to the key (if applicable).
+
+        :param key: The unprefixed key.
+        :return: The key with any configured prefix prepended.
+        """
+        pfx = self.key_prefix if self.key_prefix is not None else ''
```

Comment:
I was worried about the need to support a prefix of 0 (int) here ... maybe that is over-thinking it, but the `is not None` approach seemed safer.  Thoughts?